src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java

```diff
@@ -476,13 +536,225 @@ public DataFile parseDataFile(JsonObject datafileJson) {
+        // parse DataTable
+        JsonArray dataTablesJson = datafileJson.getJsonArray("dataTables");
         
+        if ((dataTablesJson != null ) && (!dataTablesJson.isEmpty())){
+            logger.log(Level.INFO, "dataTablesJson:size={0}", dataTablesJson.size());
+            // get parsing results of a DataTable
+            List<DataTable> dataTables = parseDataTables(dataTablesJson);
+            logger.log(Level.INFO, "dataTables:size={0}", dataTables.size());
+            logger.log(Level.FINE, "returned dataTables={0}", xstream.toXML(dataTables));
+            dataFile.setDataTables(dataTables);
+            dataFile.setDataTable(dataTables.get(0));
+            dataTables.get(0).setDataFile(dataFile);
+            dataTables.get(0).setOriginalFileFormat(originalFileFormat);
+            
+        }
+        logger.log(Level.INFO, "dataFile: parseDataFile={0}",xstream.toXML(dataFile));
         return dataFile;
     }
+    
+    
+    
+    public List<DataTable> parseDataTables(JsonArray dataTablesJson){
+        logger.log(Level.INFO, "parseDataTables is called");
+        List<DataTable> dataTables = new LinkedList<>();
+        if ((dataTablesJson !=null) && (!dataTablesJson.isEmpty())){
+            logger.log(Level.INFO, "dataTables is not empty");
+            for (JsonObject dataTableJsonL : dataTablesJson.getValuesAs(JsonObject.class)){
+                JsonObject dataTableJson = dataTableJsonL.getJsonObject("dataTable");
+                DataTable dataTable = new DataTable();
+                // capture scalar items
+                // varQuantity
+                long varQuantity =  dataTableJson.getJsonNumber("varQuantity").longValue();
+                dataTable.setVarQuantity(varQuantity);
+                // caseQuantity
+                long caseQuantity = dataTableJson.getJsonNumber("caseQuantity").longValue();
+                dataTable.setCaseQuantity(caseQuantity);
+                // UNF
+                String UNF = dataTableJson.getString("UNF", null);
+                dataTable.setUnf(UNF);
+                // call the method for pasring dataVariables array
+                List<DataVariable> dataVariables = parseDataVariables(dataTableJson.getJsonArray("dataVariables"), dataTable);
+                logger.log(Level.INFO, "returned dataVariables list: size={0}", dataVariables.size());
+                
+                dataTable.setDataVariables(dataVariables);
+                dataTables.add(dataTable);
+                logger.log(Level.INFO, " dataTables: current size={0}", dataTables.size());
+            }
+        }
+        logger.log(Level.INFO, "dataTables: size(final)={0}", dataTables.size());
+        return dataTables;
+    }
+    
+    
+    // note: the following method has not yet implemented a few of processXXX-
+    // calls in DataTableImportDDI#processVar
+    
+    public List<DataVariable> parseDataVariables(JsonArray dataVariablesJson, DataTable dataTable){
```

Comment:
I'm a little surprised all these methods are necessary? Didn't the code @lubitchv write for the Data Curation Tool already need methods like this?